addons/l10n_id_efaktur_coretax/models/account_move.py

```diff
@@ -0,0 +1,317 @@
+    l10n_id_stlg_rate = fields.Float(string="STLG Rate", default=0, copy=False)  # luxury good rate
+    l10n_id_efaktur_available = fields.Boolean(compute="_compute_l10n_id_efaktur_available")
+
+    l10n_id_efaktur_document = fields.Many2one('l10n_id_efaktur_coretax.document', readonly=True, copy=False, string="e-Faktur Document")
+
+    def _compute_need_kode_transaksi(self):
+        """ OVERRIDE: l10n_id_efaktur
+
+        By setting this l10n_id_need_kode_transaksi, we can prevent the old E-Faktur flow to be
+        triggered(i.e. efaktur range consumption).
+        """
+        self.l10n_id_need_kode_transaksi = False
+
+    @api.depends('partner_id.l10n_id_pkp', 'line_ids.tax_ids')
+    def _compute_l10n_id_efaktur_available(self):
+        """ Similar use case as l10n_id_need_kode_transaksi from l10n_id_efaktur
+
+        helps to check whether or not some fields need to be visible or not
+        """
+        for move in self:
+            move.l10n_id_efaktur_available = (
+                move.partner_id.l10n_id_pkp
+                and move.country_code == 'ID'
+                and move.move_type == 'out_invoice'
+                and move.line_ids.tax_ids
+            )
+
+    # Interactions that is triggered when choosing kode 07 or 08
+    # as they require additional information
+    @api.onchange("l10n_id_add_info_07")
+    def _onchange_l10n_id_add_info_07(self):
+        if not self.l10n_id_add_info_07:
+            self.l10n_id_facility_info_07 = False
+        else:
+            digit = self.l10n_id_add_info_07[-2:]
+            self.l10n_id_facility_info_07 = 'TD.011' + digit
+
+    @api.onchange("l10n_id_facility_info_07")
+    def _onchange_l10n_id_facility_info_07(self):
+        if not self.l10n_id_facility_info_07:
+            self.l10n_id_add_info_07 = False
+        else:
+            digit = self.l10n_id_facility_info_07[-2:]
+            self.l10n_id_add_info_07 = 'TD.005' + digit
+
+    @api.onchange("l10n_id_add_info_08")
+    def _onchange_l10n_id_add_info_08(self):
+        if not self.l10n_id_add_info_08:
+            self.l10n_id_facility_info_08 = False
+        else:
+            digit = self.l10n_id_add_info_08[-2:]
+            self.l10n_id_facility_info_08 = 'TD.011' + digit
+
+    @api.onchange("l10n_id_facility_info_08")
+    def _onchange_l10n_id_facility_info_08(self):
+        if not self.l10n_id_facility_info_08:
+            self.l10n_id_add_info_08 = False
+        else:
+            digit = self.l10n_id_facility_info_08[-2:]
+            self.l10n_id_add_info_08 = 'TD.005' + digit
+
+    def download_efaktur(self):
+        """OVERRIDE l10n_id_efaktur
+
+        Change the flow of efaktur downloading. Collects data needed for efaktur and generate the
+        xml file.
+        """
+        # Pre-download checks
+        for record in self:
+            if record.state == 'draft':
+                raise ValidationError(_('Could not download E-faktur in draft state'))
```

Comment:
1. You may want to gather all the issues and show them to the user simultaneously in a single ValidationError.
2. Try to be as straightforward as possible about the errors; at least give the record name so the user knows which record is problematic when doing it on multiple records.

Reply:
But if we gather all the issues, wouldn't it be difficult for the customers to know why they can't download the E-Faktur? (or maybe I'm misunderstanding what you mean with gather all issues lol)

Reply:
or do you mean like we compile all error messages?
```
err_message = ""
if record.state == 'draft': 
   err_message += _("Could not download E-faktur in draft state")
if ...:
  err_message += ....


if err_message:
  raise ValidationError(err_message)
```

Reply:
Yes exactly, you would show a single error, but display all the issues at once

Reply:
yeah that's a good idea, will implement that